presto-tests/src/test/java/com/facebook/presto/tests/TestQueryManager.java

```diff
@@ -278,4 +302,199 @@ public void testQueryCountMetrics()
+                TEST_SESSION.toSessionRepresentation(),
+                state,
+                new MemoryPoolId("reserved"),
+                true,
+                URI.create("1"),
+                ImmutableList.of("2", "3"),
+                query,
+                Optional.empty(),
+                Optional.empty(),
+                new QueryStats(
+                        DateTime.parse("1991-09-06T05:00-05:30"),
+                        DateTime.parse("1991-09-06T05:01-05:30"),
+                        DateTime.parse("1991-09-06T05:02-05:30"),
+                        DateTime.parse("1991-09-06T06:00-05:30"),
+                        Duration.valueOf("8m"),
+                        Duration.valueOf("5m"),
+                        Duration.valueOf("7m"),
+                        Duration.valueOf("34m"),
+                        Duration.valueOf("5m"),
+                        Duration.valueOf("6m"),
+                        Duration.valueOf("35m"),
+                        Duration.valueOf("44m"),
+                        Duration.valueOf("9m"),
+                        Duration.valueOf("10m"),
+                        Duration.valueOf("11m"),
+                        13,
+                        14,
+                        15,
+                        16,
+                        100,
+                        17,
+                        18,
+                        34,
+                        19,
+                        20.0,
+                        43.0,
+                        DataSize.valueOf("21GB"),
+                        DataSize.valueOf("22GB"),
+                        DataSize.valueOf("23GB"),
+                        DataSize.valueOf("24GB"),
+                        DataSize.valueOf("25GB"),
+                        DataSize.valueOf("26GB"),
+                        DataSize.valueOf("42GB"),
+                        true,
+                        Duration.valueOf("23m"),
+                        Duration.valueOf("24m"),
+                        Duration.valueOf("0m"),
+                        Duration.valueOf("26m"),
+                        true,
+                        ImmutableSet.of(WAITING_FOR_MEMORY),
+                        DataSize.valueOf("123MB"),
+                        DataSize.valueOf("27GB"),
+                        28,
+                        DataSize.valueOf("29GB"),
+                        30,
+                        DataSize.valueOf("32GB"),
+                        40,
+                        DataSize.valueOf("31GB"),
+                        32,
+                        33,
+                        DataSize.valueOf("34GB"),
+                        DataSize.valueOf("35GB"),
+                        DataSize.valueOf("36GB"),
+                        ImmutableList.of(),
+                        ImmutableList.of(),
+                        new RuntimeStats()),
+                Optional.empty(),
+                Optional.empty(),
+                ImmutableMap.of(),
```

Comment:
Tangential and non-blocking, but I'd like to get in the habit of using Collections.emptyMap instead of ImmutableList.of to reduce our dependence on and vulnerability to third party libraries. ImmutableMap.of hasn't been needed for years now.

Reply:
If this is the case, then we should update the contributing guide to state this: https://github.com/prestodb/presto/blob/9a3c695b1403a14e9e4b80f60920f1675115aec2/CONTRIBUTING.md?plain=1#L199